src/main/java/frc/robot/subsystems/shooter/IShooterIO.java

```diff
@@ -0,0 +1,30 @@
+package frc.robot.subsystems.shooter;
+
+import edu.wpi.first.epilogue.Logged;
+import edu.wpi.first.epilogue.Logged.Strategy;
+
+@Logged(strategy = Strategy.OPT_IN)
+public interface IShooterIO {
+
+    /**
+     * Gets the inputs from the robot to be used in the code
+     * @return ShooterIOInputs
+     */
+    @Logged(name = "Inputs", importance = Logged.Importance.CRITICAL)
+    public ShooterIOInputs getInputs();
+
+    /**
+     * Sets the outputs given by the code to be used in the robot
+     * @param outputs
+     */
+    public void setOutputs(ShooterIOOutputs outputs);
```

Comment:
```suggestion
    @Logged(name = "Outputs", importance = Logged.Importance.CRITICAL)
    public void setOutputs(ShooterIOOutputs outputs);
```

Should an `@Logged` be included here so the outputs are included in the logs? I would assume so but I noticed you did not included it in `IDrivetrainIO` so I thought I would check with you first. (Are you excluding the `@Logged` above `setOutputs` in `IDrivetrainIO` because it's outputs alone shouldn't be logged, or because every subsystem interface should do the same?)

Reply:
Actually, I noticed the Input and Output classes themselves are being logged. I think I see, `setOutputs` doesn't actual contain the values of the outputs, since it takes in the outputs as a parameter and sends them to the robot (hence there would be nothing to log). A follow up to that though, why is it that `getInputs` is logged even though it is similar to `setOutputs`? Is it because it's capturing the inputs from the robot?

Reply:
You hit the nail on the head with why `setOutputs` is not logged: it only takes the outputs as a parameter. Since it is void, it itself does not have any values to log. On the flip side, `getInputs()` returns a loggable value, and so it's able to be logged. 

That being said, I believe we need to remove auto-logging from the interfaces and IO implementations in favor of logging the inputs/outputs in the subsystem itself. The reason for this is that Epilogue is generated at compile-time, and so the IO implementation being used is not something that it knows, and therefore cannot log. This is outlined in the documentation [here](https://docs.wpilib.org/en/latest/docs/software/telemetry/robot-telemetry-with-annotations.html#caveats-and-limitations). 

If you get interested, give the full [documentation page](https://docs.wpilib.org/en/latest/docs/software/telemetry/robot-telemetry-with-annotations.html) a read. It's very interesting and useful.